umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -627,109 +409,40 @@ func (s *RedpandaService) GetHealthCheckAndMetrics(ctx context.Context, tick uin
 				IsLive:  false,
 				IsReady: false,
 			},
-			Metrics:       Metrics{},
-			Logs:          []s6service.LogEntry{},
-			UpdatedAtTick: tick,
+			RedpandaMetrics: redpanda_monitor.RedpandaMetrics{
+				Metrics:      redpanda_monitor.Metrics{},
+				MetricsState: nil,
+			},
+			Logs: []s6service.LogEntry{},
 		}
 		return s.lastStatus, nil
 	}
 
-	baseURL := "http://localhost:9644"
-	metricsEndpoint := baseURL + "/public_metrics"
-
-	// Create a client to use for our requests
-	// If it's a mock client (used in tests), use it directly
-	// Otherwise, create a new client with timeouts based on context
-	var requestClient httpclient.HTTPClient = s.httpClient
-
-	// Only create a default client if we're not using a mock client
-	if requestClient == nil {
-		requestClient = httpclient.NewDefaultHTTPClient()
+	redpandaStatus, err := s.metricsService.Status(ctx, s.filesystem, tick)
+	if err != nil {
+		return RedpandaStatus{}, fmt.Errorf("failed to get redpanda status: %w", err)
 	}
 
-	// Start metrics fetch
-	metricsFetchStart := time.Now()
-	resp, body, err := requestClient.GetWithBody(ctx, metricsEndpoint)
-	metricsFetchTime := time.Since(metricsFetchStart)
-
-	// Handle errors from metrics fetch
-	if err != nil {
-		s.lastStatus = RedpandaStatus{
-			HealthCheck: HealthCheck{
-				IsLive:  false,
-				IsReady: false,
-			},
-			Metrics:       Metrics{},
-			Logs:          logs,
-			UpdatedAtTick: tick,
-		}
-		return s.lastStatus, fmt.Errorf("failed to check metrics endpoint: %w", err)
+	if redpandaStatus.RedpandaStatus.LastScan == nil {
+		return RedpandaStatus{}, fmt.Errorf("last scan is nil")
 	}
 
 	// Create health check structure
 	healthCheck := HealthCheck{
 		// Liveness is determined by a successful response
-		IsLive: resp.StatusCode == http.StatusOK,
-		// Readiness comes from logs analysis
-		IsReady: resp.StatusCode == http.StatusOK,
+		IsLive: redpandaStatus.RedpandaStatus.IsRunning,
+		// For now this is fine.
```

Comment:
the comment is werid?

Reply:
maybe instead explan why you use two times the same status, and explain what would need to be done to make this proper